D99-D100 AADVT instrumentation (QUS_DRV.f part: per-sweep checkpoints ffdc_ckpt after X1/Y/Z, ffdc_setx, per-row/column Courant nstep hooks ffdc_nsx/ffdc_nsz). Helpers are in ATM_DRV_dynC.f.patch. Generated by diff -u against the PRISTINE QUS_DRV.f (2026-10-05); local hunks.

--- QUS_DRV.f.orig
+++ QUS_DRV.f
@@ -154,7 +154,9 @@
        DO L=1,LM
           MFLX(:,:,L) = MU(:,:,L)*(.5*DT)
        ENDDO
+      call ffdc_setx(1)
       Call AADVTX (RM,RMOM,MM,MFLX,QLIMIT,FQU)
+      call ffdc_ckpt(1,IM,JM,LM,RM,RMOM,MM)
 
 CC    mflx(:,1:jm-1,:)=MV(:,2:jm,:)*dt
 CC    mflx(:,jm,:)=0.
@@ -165,6 +167,7 @@
        if (HAVE_NORTH_POLE) mflx(:,jm,:)=0.
 
       Call AADVTY (RM,RMOM,MM,MFLX,QLIMIT,FQV)
+      call ffdc_ckpt(2,IM,JM,LM,RM,RMOM,MM)
 CC    mflx(:,:,1:lm-1)=MW(:,:,1:lm-1)*(-dt)
 CC    mflx(:,:,lm)=0.
       DO L=1,LM
@@ -175,10 +178,12 @@
          END IF
       ENDDO
       Call AADVTZ (RM,RMOM,MM,MFLX,QLIMIT)
+      call ffdc_ckpt(3,IM,JM,LM,RM,RMOM,MM)
 CC    mflx(:,:,:)=MU(:,:,:)*(.5*dt)
        DO L=1,LM
           MFLX(:,:,L) = MU(:,:,L)*(.5*DT)
        ENDDO
+      call ffdc_setx(2)
       Call AADVTX (RM,RMOM,MM,MFLX,QLIMIT,FQU)
 C****
 C**** convert from mass to concentration units
@@ -267,6 +272,7 @@
           endif
         enddo
       enddo
+      call ffdc_nsx(j,l,nstep)
       if(courmax.gt.1.) then
          write(6,*) 'aadvtx: j,l,courmax=',j,l,courmax
          ICKERR=ICKERR+1
@@ -537,6 +543,7 @@
           endif
         enddo
       enddo
+      call ffdc_nsz(i,j,nstep)
       if(courmax.gt.1.) then
          write(6,*) 'aadvtz: i,j,courmax=',i,j,courmax
          ICKERR=ICKERR+1
